### ISSUE TITLE
Hash package contents / mtime to detect cache miss

### ISSUE THREAD

**Issue by jordwalke** [jordwalke/esy#73]
Right now, if you have a local package dependency such as symlink, `esy` will just rebuild everything any time you do `esy build`. `esy` should hash the package contents to detect if a rebuild is actually necessary.

I believe for local file system dependencies `file://blah` there are other issues with not detecting changes correctly perhaps.

**Comment by andreypopp** [comment]
Should we move issues over to reasonml/esy?

**Comment by jordwalke** [comment]
Yes, perhaps I just didn't know how to do that, and wanted to avoid having them spread out over two locations (do I have to do it manually)?
